### PR TITLE
add hdfs, yarn and hive

### PR DESCRIPTION
**Pull request by tanmaykm**
This updates the package evaluator environment to include HDFS, Yarn and Hive services.
Enables running test scripts for packages Elly.jl and Hive.jl (or similar).

Ref.: https://github.com/JuliaLang/METADATA.jl/pull/6378#issuecomment-247754881

There is a separate repo [PkgEvalHadoopEnv](https://github.com/tanmaykm/PkgEvalHadoopEnv.git) with scripts that set up the environment (originally from Elly.jl and Hive.jl). If this is acceptable, I can transfer PkgEvalHadoopEnv to JuliaCI and use it for travis CI of the packages as well.

I have tested locally, restricting to only Elly.jl and Hive.jl. I didn't see a significant difference in memory consumption.

### CODE REVIEW

File: scripts/setup.sh [hunk 1]
```diff
@@ -107,6 +107,14 @@ sudo ./$AFSCRIPT --exclude-subdir --prefix=/usr/local
 export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/
 rm $AFSCRIPT
 
+#######################################################################
+# Fetch the scripts to setup hdfs, yarn and hive
+HADOOPSCRIPTDIR="/home/vagrant/hadoop"
+git clone https://github.com/tanmaykm/PkgEvalHadoopEnv.git $HADOOPSCRIPTDIR
```
Comment (tkelman) [inline scripts/setup.sh]: this should refer to a fixed tag or sha, not "latest master" which could break

**Comment by tkelman** [comment]
about how long does this take to run? seems reasonable, and would be good to move that repo to this org. do you plan on having the packages' travis setup reuse the same scripts?

**Comment by tanmaykm** [comment]
I did not measure accurately, but it was not unwieldy. (was probably around 5 minutes for both packages). I will time it with the next run.

**Comment by tanmaykm** [comment]
Updated as discussed:
- PkgEvalHadoopEnv is now in JuliaCI
- script.sh uses a tagged version of PkgEvalHadoopEnv

The tests took around 11 minutes on my laptop.
I will raise separate PRs on Elly and Hive to reuse the scripts.

**Comment by tkelman** [comment]
What did that 11 minutes include? The whole runtests modified to setup and run the tests for those couple of packages? The existing setup.sh and VM startup takes a fair amount of time and it's several hours to get through all packages, so this should be fine. Thanks!

**Comment by tanmaykm** [comment]
Yes. It was the time around [`for f in $LOOPOVER;`](https://github.com/JuliaCI/PackageEvaluator.jl/blob/73396943cfc074c3db74dbad1059086d25e106fb/scripts/setup.sh#L150) with `LOOPOVER` set to Elly and Hive.

**Comment by tkelman** [comment]
Oh okay, I wasn't so much concerned with how much time the individual packages' tests take (I doubt these are the worst offenders), but rather the setup of this download and installation in the initial provisioning of `setup.sh`.

**Comment by tanmaykm** [comment]
Ok. That's primarily a function of the internet connection speed. Nothing heavy happens beyond an untar to configure them. Sizes: 92MB for hive and 210MB for hdfs+yarn.

On Travis, Hive (which requires both of those) takes 5 minutes (https://travis-ci.org/JuliaDB/Hive.jl/jobs/163879833) for the whole process.

**Comment by tkelman** [comment]
I'm concerned this may be responsible for a number of `OutOfMemoryError`s that have started happening the last couple days. I tried giving the VM a bit more memory, but if that doesn't fix matters I may try reverting this and see if the situation improves.

**Comment by tanmaykm** [comment]
Okay. An additional 1.5 GB may be just right, from what I saw with `top`.

It would be good to have a way to start and stop services required for a package just around that package's test. That way packages won't affect each other.

**Comment by tkelman** [comment]
It takes 1.5 GB just to have these sitting in the background not doing anything? That's kind of crazy. We're running 6 VM's at the same time, I'm not sure how much total room the host system has to add much more. Adding 500 MB more per VM helped a bit and got many packages passing again, but not quite all of the new `OutOfMemoryError` cases. I'll revert this for the time being. I guess we can do something like https://github.com/JuliaCI/PackageEvaluator.jl/pull/143 if it's not appropriate for the package tests to be trying to start/stop these services in normal execution.

**Pull request merged**